worker/import.go

```diff
@@ -0,0 +1,272 @@
+/*
+ * SPDX-FileCopyrightText: © Hypermode Inc. <hello@hypermode.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package worker
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	apiv25 "github.com/dgraph-io/dgo/v250/protos/api.v25"
+	"github.com/dgraph-io/ristretto/v2/z"
+	"github.com/hypermodeinc/dgraph/v25/conn"
+	"github.com/hypermodeinc/dgraph/v25/posting"
+	"github.com/hypermodeinc/dgraph/v25/protos/pb"
+	"github.com/hypermodeinc/dgraph/v25/schema"
+
+	"github.com/dustin/go-humanize"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+// streamProcessor defines the common interface for stream processing
+type streamProcessor interface {
+	SendAndClose(*apiv25.StreamSnapshotResponse) error
+	Recv() (*apiv25.StreamSnapshotRequest, error)
+	grpc.ServerStream
+}
+
+func ProposeDrain(ctx context.Context, drainMode *pb.DrainModeRequest) ([]uint32, error) {
+	memState := GetMembershipState()
+	currentGroups := make([]uint32, 0)
+	for gid := range memState.GetGroups() {
+		currentGroups = append(currentGroups, gid)
+	}
+
+	for _, gid := range currentGroups {
+		if groups().ServesGroup(gid) && groups().Node.AmLeader() {
+			if _, err := (&grpcWorker{}).ApplyDrainmode(ctx, drainMode); err != nil {
+				return nil, err
+			}
+			continue
+		}
+		glog.Infof("[import] Connecting to the leader of the group [%v] from alpha addr [%v]", gid, groups().Node.MyAddr)
+
+		pl := groups().Leader(gid)
```

Comment:
We might set draining mode on the leader node, but the leader could change. What happens then? Right now it's only single node, so that might not be an issue for now.

Reply:
We call ApplyDrainmode on the current leader, which applies drain mode to all nodes in the group by proposing the drain mode change using the proposeAndSend function